src/components/common/cubeDimensions.js

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable */
+
+import { view } from '../../namespace.js'
+
+function toViewDimension ({
+  view,
+  cubeDimension
+}) {
+  const result = view.dimension({ cubeDimension })
+  if (!result) {
+    const cubeDimensionTerm = cubeDimension.ptr?cubeDimension.ptr.term:cubeDimension
+    // Try to fetch using view:as
+    const candidate =  view.ptr.node(cubeDimensionTerm).in(view.as).term
+    if (!candidate){
+      throw Error('No dimension found for ', cubeDimensionTerm)
+    }
+    for (const dimension of view.dimensions){
+      if (dimension.ptr.term.equals(candidate)) {
+        return dimension
+      }
+    }
+  }
+  return result
+}
+
+function cubeDimensionsWithFallBack ({ dimension }) {
```

Comment:
get the cubeDimensions that correspond to a viewDimension. There are cases where we don't count with cubeDimensions by the library, in those cases this adds placeholders to be used in the app.